packages/api/src/platforms/vtex/resolvers/productGroup.ts

```diff
@@ -1,19 +1,16 @@
 import { enhanceSku } from '../utils/enhanceSku'
-import type { Product } from '../clients/search/types/ProductSearchResult'
 import type { Resolver } from '..'
+import type { PromiseType } from '../../../typings'
+import type { StoreProduct } from './product'
 
-export const StoreProductGroup: Record<string, Resolver<Product>> = {
-  hasVariant: (root) => root.skus.map((sku) => enhanceSku(sku, root)),
-  productGroupID: ({ product }) => product,
-  name: ({ name }) => name,
-  additionalProperty: ({ textAttributes = [], productSpecifications = [] }) => {
-    const specs = new Set(productSpecifications)
+type Root = PromiseType<ReturnType<typeof StoreProduct.isVariantOf>>
 
-    return textAttributes
-      .filter((attribute) => specs.has(attribute.labelKey))
-      .map((attribute) => ({
-        name: attribute.labelKey,
-        value: attribute.labelValue,
-      }))
-  },
+export const StoreProductGroup: Record<string, Resolver<Root>> = {
+  hasVariant: (root) =>
+    root.isVariantOf.items.map((item) => enhanceSku(item, root.isVariantOf)),
+  productGroupID: ({ isVariantOf }) => isVariantOf.productId,
+  name: ({ isVariantOf }) => isVariantOf.productName,
+  // TODO: Check if textAttribute is another property from search
+  additionalProperty: ({ isVariantOf: { skuSpecifications = [] } }) =>
```

Comment:
we use textAttributes to [build `skuSpecifications`](https://github.com/vtex/vtexis-compatibility-layer/blob/main/src/convertISProduct.ts#L110) and [`productClusters`](https://github.com/vtex/vtexis-compatibility-layer/blob/main/src/convertISProduct.ts#L247). so you will already have `skuSpecifications` here

Reply:
Does the labelKey and labelValue exist in this skuSpecification type?

Reply:
it will have something like this:
```
skuSpecifications: [
  {
    field: {
      name: 'Color',
      originalName: 'Color',
    },
    values: [
      {
        name: 'Black',
        originalName: 'Black',
      },
    ],
  },
],
```
so I think `labelKey` would be field.name and `labelValue` would be value.name for each value

Reply:
@tlgimenes i need your help here. What values are expected here?